src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -136,12 +137,71 @@ public ViewMode CurrentMode
         }
 
         /// <summary>
-        ///     SearchResults property
+        ///     Items that were found during search.
         /// </summary>
-        /// <value>
-        ///     This property is observed by SearchView to see the search results
-        /// </value>
-        public ObservableCollection<NodeSearchElementViewModel> SearchResults { get; private set; }
+        private List<NodeSearchElementViewModel> searchResults;
+
+        private IEnumerable<NodeSearchElementViewModel> filteredResults;
+        /// <summary>
+        /// Filtered search results.
+        /// </summary>
+        public IEnumerable<NodeSearchElementViewModel> FilteredResults
+        {
+            set
+            {
+                filteredResults = value;
+                RaisePropertyChanged("FilteredResults");
+            }
+            get
+            {
+                return filteredResults;
+            }
+        }
+
+        /// <summary>
+        /// Filters search items, if category was selected.
+        /// </summary>
+        internal void Filter()
+        {
+            var allowedCategories = SearchCategories.Where(cat => cat.IsSelected);
+            FilteredResults = searchResults.Where(x => allowedCategories
+                                                                       .Select(cat => cat.Name)
+                                                                       .Contains(x.Category));
+        }
+
+        /// <summary>
+        /// Returns true, if it was found at least one item. Otherwise it returns false.
+        /// </summary>
+        public bool IsAnySearchResult
+        {
+            get
+            {
+                return searchResults.Any();
+            }
+        }
+
+        private IEnumerable<SearchCategory> searchCategories;
+        /// <summary>
+        /// Categories that were found after search. Used to filter search results.
+        /// </summary>
+        public IEnumerable<SearchCategory> SearchCategories
+        {
+            get
+            {
+                return searchCategories;
+            }
+            private set
+            {
+                // Unsubscribe old categories.
+                for (int i = 0; i < searchCategories.Count(); i++)
```

Comment:
I dont think this is necessary. Because you are going to set new values, in that case old values will become obsolete. 
That case, why there is a need to unsubscribe?